### DISCUSSION TITLE
Unable to export a drum kit

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1839] [Hydrogen on Windows]
I can no longer back up a drum kit file. In the past, I was able to export a kit as a file with the *.drumkit" file extension. Now I get the following erroneous message: 

"You do not have permissions to write to the selected folder. Please select another one."

I tried exporting to a folder I created. Any way to correct this issue without re-installing the program?

**Comment by trebmuh** [comment]
Hi @Djard 

Which version of Hydrogen are you using?
Which version of the operating system?

**Comment by theGreatWhiteShark** [comment]
Hey @Djard ,

Well, this error occurs because Windows is telling Hydrogen it does not have write permissions for the folder you chose. This does not sound like a problem within the Hydrogen code?

Which folder did you selected? And are you able to e.g. save the song in it or any other data with another application? Is something showing up in the log?

**Reply by satur-9** [reply]
I also have this problem, but all other save and export functions appear to work. i also tried running as administrator and same behaviour. definitely a hydrogen problem, or a hydrogen on windows problem. but not _just_ a windows problem

**Reply by theGreatWhiteShark** [reply]
Could you tell which OS and Hydrogen version you are using and attach the [log](https://github.com/hydrogen-music/hydrogen/wiki/Accessing-log-messages-of-Hydrogen)? When we are able to reproduce the problem it will be much more easy to fix.

**Reply by cme** [reply]
> I also have this problem, but all other save and export functions appear to work. i also tried running as administrator and same behaviour. definitely a hydrogen problem, or a hydrogen on windows problem. but not _just_ a windows problem

If you could set the log level to Debug (using the Debug menu) and share the tail of the log here that might be a great help!!

**Reply by satur-9** [reply]
thanks for the fast response, log at end
contingent success! from what i can see the dialog seems to be confused between what is a file and what is a directory and is messing up the trailing slash. i tried appending a filetype to the filename to see if that would help but it didn't, then i tried adding a / and that worked but in the confirmation message you can see there a // on the end. so there is a workaround but the behaviour is confusing for sure
![hydrogen export kit issue](https://github.com/hydrogen-music/hydrogen/assets/28830464/04998b83-343d-43a5-b091-81c700040543)

here you go, windows 7, hydrogen 1.2.2:
(E) PortAudioDriver::connect Can't use device '', using default stream[0m
(E) PortAudioDriver::connect Can't use device '', using default stream[0m
(E) Filesystem::check_permissions C:/Users/Matty/Desktop/Deathclaw Bathroom Drums is not a directory[0m
(E) Filesystem::check_permissions C:/Users/Matty/Desktop/Deathclaw Bathroom Drums is not a directory[0m
(E) Filesystem::check_permissions C:/Users/Matty/Desktop/Deathclaw Bathroom Drums is not a directory[0m
(E) Filesystem::check_permissions C:/Users/Matty/Desktop/DeathClaw Samples/Hydrogen Drumkits/Deathclaw Bathroom Drums.Drumkit is not a directory[0m
(E) Filesystem::check_permissions C:/Users/Matty/Desktop/DeathClaw Samples/Hydrogen Drumkits/Deathclaw Bathroom Drums.Drumkit is not a directory[0m
(I) Drumkit::exportTo Export drumkit to [C:/Users/Matty/Desktop/DeathClaw Samples/Hydrogen Drumkits//Deathclaw_Bathroom_Samples_Kit.h2drumkit] using the most recent format[0m

**Reply by theGreatWhiteShark** [reply]
Thanks a lot! I can reproduce it now. ~It is indeed a whitespace issue.~

**Reply by theGreatWhiteShark** [reply]
@satur-9 could you check whether the problem still occurs in [this version](https://ci.appveyor.com/api/buildjobs/p4gmuw74a72ssumr/artifacts/Hydrogen-1.2.2-111-g18dfa5c63-win64.exe)?

**Reply by satur-9** [reply]
yep, drumkit export dialogue now functions as expected - perfect *chefs kiss*. exported a kit no problem, and was able to load and open it. fantastic work!

**Reply by theGreatWhiteShark** [reply]
Awesome. Thanks a lot for testing!
